department-of-veterans-affairs/caseflow
# Wmedders21/appeals 43185

Resolves [Create a New Route for Work Order Packages - traced to req Appeals-31793](https://jira.devops.va.gov/browse/APPEALS-43185)

# Description
Please explain the changes you made here.

1. Adds migration and model for join table to set up many-to-many relationship between `TranscriptionPackage` and `Hearings`
2. Adds migration and model for join table to set up many-to-many relationship between `TranscriptionPackage` and `LegacyHearings`
3. Adds new endpoint `hearings/transcription_packages/:task_number`
4. Adds a custom serializer `TranscriptionPackageSerializer` that returns necessary data used for the Transcription Dispatch table
5. Updates the `TranscriptionPackage` factory to have more realistic values
6. Adds new methods to `TranscriptionPackage` model to house logic for new serializer

## Acceptance Criteria

## Testing Plan

1. Go to [Jira Issue/Test Plan Link](https://jira.devops.va.gov/browse/JIRA-12345) or list them below

_The following steps assume the tester has run pending migrations, and seeds or `make reset`_
1. In a terminal, navigate to caseflow root directory
2. Run `rails c` to enter development console
3. Copy and paste the following data setup script into the console and press `enter`
```
pry_instance.config.pager = false
hearings = Hearing.limit(5)
legacy_hearings = LegacyHearing.limit(5)
transcription_package = FactoryBot.create(:transcription_package)
hearings.each do |hearing|
  TranscriptionPackageHearing.create!(
    hearing_id: hearing.id,
      transcription_package_id: transcription_package.id
    )
end
legacy_hearings.each do |legacy_hearing|
  TranscriptionPackageLegacyHearing.create!(
    legacy_hearing_id: legacy_hearing.id,
    transcription_package_id: transcription_package.id
  )
end
transcription_package.update!(
  contractor_id: TranscriptionContractor.first&.id,
  status: "Successfully uploaded to Box.com"
)
```
4. In another terminal window, run caseflow locally
5. In Chrome, navigate to `http://localhost:3000/hearings/transcription_packages/BVA-1111-0001`
6. Click the "pretty print" checkbox
7. Expect to see a valid JSON response similar to the screenshot below
![Screenshot 2024-05-29 at 7 00 27 PM](https://github.com/department-of-veterans-affairs/caseflow/assets/93014155/14b9c48e-c483-4f6a-9dd9-173e32262c07)



# Frontend
## User Facing Changes

 BEFORE|AFTER
 ---|---

## Storybook Story
*For Frontend (Presentation) Components*

# Backend
## Database Changes
*Only for Schema Changes*


## Integrations: Adding endpoints for external APIs

# Best practices
## Code Documentation Updates

## Tests
### Test Coverage
Did you include any test coverage for your code? Check below:

### Code Climate
Your code does not add any new code climate offenses? If so why?

## Monitoring, Logging, Auditing, Error, and Exception Handling Checklist

### Monitoring

### Logging

### Auditing

### Error Handling

### Exception Handling
